pg_shard--1.0--1.1.sql

```diff
@@ -12,7 +12,16 @@ AS $sync_table_metadata_to_citus$
 		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
 		dummy_shard_length CONSTANT bigint := 0;
 	BEGIN
-		-- copy shard placement metadata
+		-- grab lock for upsert
+		LOCK TABLE pg_dist_shard_placement IN EXCLUSIVE MODE;
```

Comment:
The [StackOverflow answer](http://stackoverflow.com/a/17267423) I was referencing used `EXCLUSIVE MODE` and I tend to agree with it. If the CitusDB metadata tables are out of sync with the `pg_shard` ones, then any queries processed by CitusDB will potentially hit stale shards, so one could make the argument that we should lock them out until the update is complete.

But because right now this sync function requires manual intervention anyways, they will _always_ read stale metadata until a human runs this function, so I think it's fine to let them go ahead and keep reading metadata while we are updating it. We just need to ensure the upsert itself is serialized, which is what the lock provides.